src/main/java/org/folio/des/service/impl/JobServiceImpl.java

```diff
@@ -96,6 +102,19 @@ public JobCollection get(Integer offset, Integer limit, String query) {
   @Transactional
   @Override
   public org.folio.des.domain.dto.Job upsertAndSendToKafka(org.folio.des.domain.dto.Job jobDto, boolean withJobCommandSend) {
+    Optional.ofNullable(jobDto.getExportTypeSpecificParameters()).ifPresent(
```

Comment:
You can replace chain of "ifPresent" to map method

Reply:
Are you suggesting like below...

**Optional.ofNullable(jobDto.getExportTypeSpecificParameters())
     .map(ExportTypeSpecificParameters::getVendorEdiOrdersExportConfig)
       .map(VendorEdiOrdersExportConfig::getExportConfigId).ifPresent(configId->{
              try {
                log.info("Looking config with id {}", configId.toString());
                client.getConfigById(configId.toString());
              } catch (NotFoundException e) {
                log.info("config not found", configId.toString());
                jobDto.getExportTypeSpecificParameters().getVendorEdiOrdersExportConfig().getEdiSchedule().
                   getScheduleParameters().setSchedulePeriod(ScheduleParameters.SchedulePeriodEnum.NONE);
                throw new NotFoundException(String.format(INTEGRATION_NOT_AVAILABLE, configId));
      }});**

Reply:
yes, that

Reply:
done!